File: Data Structures & Algorithms/generate-parentheses/submission-0.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        res = []
        self.genPar(n, "", 0, 0, res)
        return res
    
    def genPar(self, n, stack, op, cl, res):
        if op == n and cl == n:
            res.append(stack)
        
        a, b = None, None
        if op < n:
            self.genPar(n, stack + '(', op + 1, cl, res)
        if cl < n and cl < op:
            self.genPar(n, stack + ')', op, cl + 1, res)

        return a, b

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<>();
        int[] res = new int[2];

        for (int i = 0; i < nums.length; i++) {
            hm.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            int need = target - nums[i];

            if (hm.containsKey(need) && hm.get(need) != i) {
                int found = hm.get(need);
                res[0] = Math.min(i, found);
                res[1] = Math.max(i, found);
            }
        }

        return res;
    }
}
